packages/api/src/platforms/vtex/resolvers/query.ts

```diff
@@ -273,4 +274,22 @@ export const Query = {
       address,
     }
   },
+  redirect: async (
+    _: unknown,
+    { term, selectedFacets }: QueryRedirectArgs,
+    ctx: Context
+  ) => {
+    if (!term && (!selectedFacets || !selectedFacets.length)) {
+      return null
+    }
+
+    const { redirect } = await ctx.clients.search.products({
```

Comment:
@icazevedo should this use the `productSearchPromise`?

Reply:
I think you are right, @Git-the-Sanz. You don't have access to the `productSearchPromise` here.